src/number_theory/primorial.lean

```diff
@@ -101,15 +106,15 @@ begin
     exact mul_dvd_mul_left a step, }
 end
 
-lemma primorial_le_pow_4 : ∀ (n : ℕ), ∏ p in (filter prime (range (n + 1))), p ≤ 4 ^ n
+lemma primorial_le_pow_4 : ∀ (n : ℕ), n# ≤ 4 ^ n
 | 0 := le_refl _
 | 1 := le_of_inf_eq rfl
 | (n + 2) :=
 begin
-  cases nat.mod_two_eq_zero_or_one (n + 1),
-  { rcases (nat.even_iff.2 h) with ⟨m, twice_m⟩,
-    calc (∏ i in filter prime (range (n + 3)), i)
-          = ∏ i in filter prime (range (2 * m + 2)), i : by rw ←twice_m
+  cases nat.mod_two_eq_zero_or_one (n + 1) with n_odd n_even,
```

Comment:
I would expect `n_odd` and `n_even` to be swapped...

Reply:
This is because the actual types are "n + 1 is even" and "n + 1 is odd". I could go with e.g. `sn_even` and `sn_odd`, I suppose.

Reply:
Ooh, you're totally right. No, it's fine this way.